src/sonic-yang-models/yang-models/sonic-port.yang

```diff
@@ -94,6 +94,12 @@ module sonic-port{
 						pattern "on|off";
 					}
 				}
+
+				leaf autoneg {
+					type uint16 {
+						range 0..256;
```

Comment:
why is autoneg is a bool? on/off?

Reply:
Question is: why autoneg is not the bool? also if you have a pointer towards the design document of this feature, which explains the limit of 256, that will help. Thx.

Reply:
@lguohan @praveen-li 
Yang model depends on config_db, so if in config_db we have integer, than it must be in yang model in another case we will have error and crush.
We can see in portsorch code that string value from config_db convert to integer:
https://github.com/Azure/sonic-swss/blob/189a9641cf47e81c7c90d2501e70b73b74b082e1/orchagent/portsorch.cpp#L2215-L2219
```
    /* Set autoneg and ignore the port speed setting */
    if (fvField(i) == "autoneg")
    {
        an = (int)stoul(fvValue(i));
    }
```

In next step, we can see how this integer value converts to bool type and use as sai attribute:
https://github.com/Azure/sonic-swss/blob/189a9641cf47e81c7c90d2501e70b73b74b082e1/orchagent/portsorch.cpp#L1752-L1767
```
bool PortsOrch::setPortAutoNeg(sai_object_id_t id, int an)
{
    SWSS_LOG_ENTER();

    sai_attribute_t attr;
    attr.id = SAI_PORT_ATTR_AUTO_NEG_MODE;
    switch(an) {
      case 1:
        attr.value.booldata = true;
        break;
      default:
        attr.value.booldata = false;
        break;
    }

    sai_status_t status = sai_port_api->set_port_attribute(id, &attr);
```


So in config_db we need use integer for valid conversion to int and after that int to bool. So in yang models we need use int value too.

@praveen-li , I agree with you that we no need use range 0..255. We must use only 0 or 1. Thanks for the correction.